Source/WebCore/page/Navigator.cpp

```diff
@@ -247,11 +267,44 @@ void Navigator::initializePluginAndMimeTypeArrays()
 
     auto [publiclyVisiblePlugins, additionalWebVisiblePlugins] = frame->page()->pluginData().publiclyVisiblePluginsAndAdditionalWebVisiblePlugins();
 
+    m_pdfViewerEnabled = frame->loader().client().canShowMIMEType("application/pdf"_s);
+
+    // macOS uses a PDF Plugin (which may be disabled). Other ports handle PDF's through native
+    // platform views outside the engine, or use pdf.js.
+    bool hasEnabledWebKitBuiltinPDFPlugin = frame->page()->pluginData().builtInPDFPlugin().has_value();
+
     Vector<Ref<DOMPlugin>> publiclyVisibleDOMPlugins;
     Vector<Ref<DOMPlugin>> additionalWebVisibleDOMPlugins;
     Vector<Ref<DOMMimeType>> webVisibleDOMMimeTypes;
 
-    publiclyVisibleDOMPlugins.reserveInitialCapacity(publiclyVisiblePlugins.size());
+    size_t publiclyVisibleDOMPluginCount = publiclyVisiblePlugins.size();
+#if PLATFORM(COCOA) || ENABLE(PDFJS)
+    if (m_pdfViewerEnabled)
+        publiclyVisibleDOMPluginCount += dummyPDFPluginCount;
+#endif
+
+    publiclyVisibleDOMPlugins.reserveInitialCapacity(publiclyVisibleDOMPluginCount);
+
+#if PLATFORM(COCOA) || ENABLE(PDFJS)
```

Comment:
There is something odd here. You set `m_pdfViewerEnabled` to true potentially earlier.
The spec says that if m_pdfViewerEnabled is true, then the plugin list must contain the 5 dummy plugin names.
However, adding the dummy plugin names is behind a `#if PLATFORM(COCOA) || ENABLE(PDFJS)` check.

Therefore, non-Cocoa ports that return true for frame->loader().client().canShowMIMEType("application/pdf"_s) will not be standard-compliant since they will return true for `navigator.pdfViewerEnabled` but not have the dummy plugins names.

Reply:
I'm not sure if we have a port that is non-Cocoa and also non-PDF.JS, but if we did it would not behave properly with my logic. I'll remove the compile guards.